ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+const axios = require('axios');
+const cheerio = require('cheerio');
+const time = require('console');
+const showdown = require('showdown');
+
+
+/** Fetches the repo owner and name for each repo on the GitHub trending page
+ * @param {number} timeMode 0: daily; 1: weekly; 2: monthly
+ * @returns {string[]} an array that stores alternatingly the owner and the name of each repo
+ */
+async function fetchRepos( timeMode: number) {
+  try {
+    const timeModes = ['daily', 'weekly', 'monthly'];
+    const response = await axios.get('https://github.com/trending?since=' + timeModes[timeMode]);
+    const $ = cheerio.load(response.data);
+    const repos: string[] = [];
+
+    $('h2 a').each((i: number, el: cheerio.Element) => {
```

Comment:
h2 is to static. There would be new h2 tags if the design of page changed. Please come up with a new idea. May be id of html tags can be used or an api can be used.

Reply:
The issue is that Github's API doesn't provide any information about its trending page. 
Therefore all the data wanted has to be received through scraping the HTML version of the file.

Reply:
I'be looked at the HTML code of the trending page for a bit but I am unsure which id/classname/tag would be the correct to choose from. Could you maybe have a look at it and tell me on which one to build from so that I can implement it here?

Reply:
@DogukanGun would something like '$$('ol.repo-list li h3').forEach(el => console.log(el.innerText));' this work better as a way too extract repos and owners. Because that's something I could definitely implement